src/index.ts

```diff
@@ -70,20 +70,22 @@ export function getType(tpe: Tpe, owner: Tpe | null): Reader<Ctx, gen.TypeRefere
       case 'Unit':
         return reader.of(gen.customCombinator('void', `${prefix}VoidFromUnit`));
       case 'Option':
-        const innerType = getType(tpe.args![0], tpe);
-        if (owner && ['List', 'Set', 'TreeSet', 'Map'].includes(owner.name)) {
-          return innerType.map(t => gen.unionCombinator([t, gen.nullType]));
-        }
-        return innerType;
+        return getType(tpe.args![0]).map(t =>
+          gen.customCombinator(
+            `Option<${gen.printStatic(t)}>`,
+            `createOptionFromNullable(${gen.printRuntime(t)})`,
+            t.kind === 'Identifier' ? [t.name] : undefined
```

Comment:
These are the dependencies, right? Not sure about this, what if you have 

`Option[List[User]]`

then `User` should be a dependency.

Reply:
yes, I'll try to add a test for this

Reply:
Re-reading this, actually my assumption was that there's should be a dependency if the model is not a primitive io-ts type, but something that has been defined already. Thus the "identifier" check. I.e in your example, User would be an identifier and thus added as the only dependency. Are thee other cases to handle?

Reply:
@gabro last commit should address this: I didn't manage to create a failing test, but looking at the diff for the `Id` generic type (I applied the same logic to this case too), now it is correctly placed before any usage (it was placed later on in the file but still it didn't cause any issue in the output since it is writing a `function Id()` - not a `const Id = () => {}` - so it gets hoisted anyway.